llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4780,7 +4780,10 @@ bool LoopVectorizationPlanner::isMoreProfitable(
   // Assume vscale may be larger than 1 (or the value being tuned for),
   // so that scalable vectorization is slightly favorable over fixed-width
   // vectorization.
-  bool PreferScalable = A.Width.isScalable() && !B.Width.isScalable();
+  bool PreferScalable = false;
+  if (!TTI.preferFixedIfEqualToScalable())
+    PreferScalable = A.Width.isScalable() && !B.Width.isScalable();
```

Comment:
This may be better as 
```
if (!TTI.preferFixedIfEqualToScalable())
  PreferScalable = !A.Width.isScalable() && B.Width.isScalable();
else
  PreferScalable = A.Width.isScalable() && !B.Width.isScalable();
```
This way it will not be dependent upon the order it visits the vplans.

PreferScalable could then do with a better name too.

Reply:
I have reverted back to my initial implementation, because your suggestion is having a wider impact that I would like to look at in a follow up because I do agree it would make sense.